api/capi/src/nnstreamer-capi-single-new.c

```diff
@@ -402,13 +407,82 @@ int
 ml_single_invoke (ml_single_h single,
     const ml_tensors_data_h input, ml_tensors_data_h * output)
 {
-  /**
-   * @todo:
-   * Setup input and output buffer
-   * Output buffer
-   * Do invoke
-   * return result
-   */
+  ml_single *single_h;
+  ml_tensors_data_s *in_data, *result;
+  GstTensorMemory in_tensors[NNS_TENSOR_SIZE_LIMIT];
+  GstTensorMemory out_tensors[NNS_TENSOR_SIZE_LIMIT];
+  GTensorFilterSingleClass *klass;
+  int i, status = ML_ERROR_NONE;
+
+  check_feature_state ();
+
+  if (!single || !input || !output) {
+    ml_loge ("The given param is invalid.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  single_h = (ml_single *) single;
+  in_data = (ml_tensors_data_s *) input;
+  *output = NULL;
+
+  if (!single_h->filter) {
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  /* Validate input data */
+  if (in_data->num_tensors != single_h->in_info.num_tensors) {
+    ml_loge ("The given param input is invalid, \
+        different number of memory blocks.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  for (i = 0; i < in_data->num_tensors; i++) {
+    size_t raw_size = ml_tensor_info_get_size (&single_h->in_info.info[i]);
+
+    if (!in_data->tensors[i].tensor || in_data->tensors[i].size != raw_size) {
```

Comment:
What if the model may accept different input sizes and the API user knows it and wants to enter various input dimensions?

Reply:
Do you mean that user wants to set different input dimension for each invoke? if thats the case, a new function can be added to update input and output info, which can be called before invoke (in a different PR I think).

Reply:
There are models that have flexible input dimensions. The current version of tensorflow-lite doesn't, though. E.g., many of our custom-filter examples have flexible input dimensions as well.

Reply:
Anyway.. this is not for 5.5 M2, but for 6.0+; thus we may discuss updating (preferably adding more semantics to it) the API.

Reply:
Please add an issue item for "Flexible Input Dimension for Single-Shot API". I think we may address this issue later independently.

Reply:
Added #1651